client/src/main/java/org/folio/metastorage/client/Client.java

```diff
@@ -243,13 +250,31 @@ private void sendChunk(ReaderProxy reader, Promise<Void> promise) {
             System.out.println(request);
             vertx.runOnContext(x -> sendChunk(reader, promise));
           } else {
-            webClient.putAbs(headers.get(XOkapiHeaders.URL) + "/meta-storage/records")
-                .putHeaders(headers)
-                .expect(ResponsePredicate.SC_OK)
-                .expect(ResponsePredicate.JSON)
-                .sendJsonObject(request)
-                .onFailure(promise::fail)
-                .onSuccess(x -> sendChunk(reader, promise));
+            if (compress) {
+              AsyncCodec.compress(vertx, request.toBuffer())
+                  .onFailure(e -> {
+                    log.error(e.getMessage(), e);
+                    promise.complete();
+                  })
+                  .onSuccess(b -> {
+                    webClient.putAbs(headers.get(XOkapiHeaders.URL) + "/meta-storage/records")
+                        .putHeaders(headers)
+                        .putHeader(HttpHeaders.CONTENT_ENCODING.toString(), "gzip")
+                        .expect(ResponsePredicate.SC_OK)
+                        .expect(ResponsePredicate.JSON)
+                        .sendBuffer(b)
+                        .onFailure(promise::fail)
+                        .onSuccess(x -> sendChunk(reader, promise));
+                  });
```

Comment:
```suggestion
            if (compress) {
              AsyncCodec.compress(vertx, request.toBuffer())
                  .compose(b ->
                      webClient.putAbs(headers.get(XOkapiHeaders.URL) + "/meta-storage/records")
                          .putHeaders(headers)
                          .putHeader(HttpHeaders.CONTENT_TYPE.toString(), "application/json")
                          .putHeader(HttpHeaders.CONTENT_ENCODING.toString(), "gzip")
                          .expect(ResponsePredicate.SC_OK)
                          .expect(ResponsePredicate.JSON)
                          .sendBuffer(b))
                  .onFailure(promise::fail)
                  .onSuccess(x -> sendChunk(reader, promise));;    
                  });
```

Reply:
Suggested compose rather than double onSuccess/onFailure.. sendJsonObject automatically sets Content-Type:application/json.. sendBuffer does not, so we have to set that.